crates/sui-json-rpc/src/authority_state.rs

```diff
@@ -565,6 +565,41 @@ impl<S: ?Sized + StateRead> ObjectProvider for Arc<S> {
     }
 }
 
+#[async_trait]
+impl<S: ?Sized + StateRead> ObjectProvider for (Arc<S>, Arc<TransactionKeyValueStore>) {
```

Comment:
Maybe we should instead implement this on TransactionKeyValueStore directly (and the FallBackStore) similar to how we handle the other interactions with the KVstore

Reply:
`get_object` method itself is implemented for TransactionKeyValueStore/FallBackStore the same way all other methods are implemented.
This is an extra trait `ObjectProvider` used by `balance_changes`. In general, `ObjectProvider` can't be implemented for every `TransactionKeyValueStoreTrait`( `find_object_lt_or_eq_version` is not available in dynamodb).
let me know if you actually meant something else:)

Reply:
ah got it, that makes more sense then. Thanks for the clarification!